drivers/timer/stm32_lptim_timer.c

```diff
@@ -274,15 +310,30 @@ u32_t z_timer_cycle_get_32(void)
 	 * of the LPTIM_CNT register, two successive read accesses
 	 * must be performed and compared
 	 */
-
 	while (lp_time != LL_LPTIM_GetCounter(LPTIM1)) {
 		lp_time = LL_LPTIM_GetCounter(LPTIM1);
 	}
+
+	/* In case of counter roll-over, add this value,
+	 * even if the irq has not yet been handled
+	 */
+	if ((LL_LPTIM_IsActiveFlag_ARRM(LPTIM1) != 0)
+	  && LL_LPTIM_IsEnabledIT_ARRM(LPTIM1) != 0) {
+		lp_time += LL_LPTIM_GetAutoReload(LPTIM1) + 1;
+	}
+
 	lp_time += accumulated_lptim_cnt;
 
+	/* convert lptim count in a nb of hw cycles with precision */
+#if defined(CONFIG_STM32_LPTIM_CLOCK_LSI)
+	u64_t ret = lp_time * ((sys_clock_hw_cycles_per_sec() / 1000)
+		/ (LPTIM_CLOCK / 1000));
+#else
+	u64_t ret = lp_time * (sys_clock_hw_cycles_per_sec() / LPTIM_CLOCK);
+#endif /* CONFIG_STM32_LPTIM_CLOCK_LSI */
```

Comment:
Why it's different for LSI?
Here, I think you want to do:
```u64_t ret = (lp_time * sys_clock_hw_cycles_per_sec()) / LPTIM_CLOCK;```

Reply:
simplification to have the same return value in case of LSI or LSE